runbotics/runbotics-orchestrator-ui/src/src-app/components/ProcessVariablesPanel/VariablesPanel/VariablesPanel.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import {
+    Box, Typography, Chip, Grid, Divider, IconButton
+} from '@mui/material';
+
+import { useTheme } from '@mui/material/styles';
+
+
+import If from '#src-app/components/utils/If';
+import useProcessVariables from '#src-app/hooks/useProcessVariables';
+import { translate } from '#src-app/hooks/useTranslations';
+
+import MenuCopy from '../MenuCopy';
+import { GridContainer, GridTag, GridVariable
+} from './VariablesPanel.styles';
+
+export enum VariableTag {
+    Variable = 'VariableTag',
+    ActionOutput = 'ActionOutputTag',
+}
+
+const VariablesPanel = () => {
+    const theme = useTheme();
+    const { globalVariables, inputActionVariables, outputActionVariables, attendedVariables } = useProcessVariables();
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const [showMenu, setShowMenu] = useState(null);    
+    
+    const [menuId, setMenuId] = useState(false);
+
+    const allProcessVariables = [...globalVariables, ...inputActionVariables, ...outputActionVariables, ...attendedVariables];
+
+    if (allProcessVariables.length === 0) {
+        return (
+            <Typography sx={{display: 'flex', justifyContent: 'center', paddingTop: '2rem'}}>
+                {translate('Process.Modeler.VariablesPanel.Empty.Message')}
+            </Typography>);
+    }
+
+    const getTagBgColor = (tag: VariableTag) => {
+        if (tag === VariableTag.Variable) {
+            return theme.palette.tag.variable;
+        } 
+        return theme.palette.tag.action;
+         
+    };
+    
+    const handleMenuClick = (event, name) => {
+        event.preventDefault();
+        setAnchorEl(event.currentTarget);
+        setShowMenu(true);
+        setMenuId(name);
+    };
+
+    const handleMenuClose = () => {
+        setAnchorEl(null);
+        setMenuId(null);
+    };
+
+    const getJSXForVariable = (name: string, tag: VariableTag) => (
```

Comment:
I would like to see this as an FC, maybe VariableRow/VariableBox/ or something like this